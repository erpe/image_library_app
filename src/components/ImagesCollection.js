import React from 'react'
import PropTypes from 'prop-types'
import ImageCard from './ImageCard'

const ImagesCollection = (props) => {
  let renderable

  if (props.displayStyle === 'card') {
    renderable = props.images.map((img) => {
      return <ImageCard image={img} />
    })
  }

  return(
    <div className='row'>
      {renderable}
    </div>
  )
}

ImagesCollection.propTypes = {
  displayStyle: PropTypes.string.isRequired,
  images: PropTypes.array.isRequired,
}

export default ImagesCollection
