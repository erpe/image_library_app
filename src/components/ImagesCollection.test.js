import React from 'react'
import { render, asFragment } from '@testing-library/react'
import ImagesCollection from './ImagesCollection'


test('renders ImagesCollection', () => {
  const {asFragment} = render(<ImagesCollection images={[]} displayStyle={'card'} />)
  expect(asFragment()).toMatchSnapshot()
})
