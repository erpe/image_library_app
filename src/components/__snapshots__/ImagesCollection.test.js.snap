// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImagesCollection 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form-inline mb-3"
    >
      <button
        class="btn btn-sm btn-outline-light mr-3"
      >
        List as table
      </button>
      <label
        class="label mr-1"
        for="category"
      >
        Filter by
      </label>
      <select
        class="form-control form-control-sm bg-dark text-light mr-3"
        name="category"
      >
        <option
          value="reset"
        >
           -- Select Category --
        </option>
      </select>
      <label
        class="mr-1"
        for="sort"
      >
        Sort by
      </label>
      <select
        class="form-control form-control-sm bg-dark text-light"
      >
        <option
          value="reset"
        >
          -- Select Sort --
        </option>
        <option
          value="alt"
        >
          Alt
        </option>
        <option
          value="category"
        >
          Category
        </option>
        <option
          value="copyright"
        >
          Copyright
        </option>
      </select>
    </div>
    <div
      class="card-columns"
    />
  </div>
</DocumentFragment>
`;
